src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -237,6 +266,11 @@ class JSONReader
             RAVELOG_ERROR_FORMAT("Failed to load env referenceUri='%s' from file '%s'", referenceUri%_filename);
         }
 
+        //If remote URL is provided, start the process to download everything and load it into the json map
+        if(IsRemote()) {
+            _DownloadRecursively(rEnvInfo);
```

Comment:
Isn't this too late? `_GetDocumentFromFilename` is already called before this line.